src/corecel/OpaqueId.hh

```diff
@@ -137,6 +145,26 @@ operator-(OpaqueId<V, S> id, std::make_signed_t<S> offset)
     return OpaqueId<V, S>{id.unchecked_get() - static_cast<S>(offset)};
 }
 
+//! Multiply an opaque ID by a factor
+template<class V, class S>
+inline CELER_FUNCTION OpaqueId<V, S>
+operator*(OpaqueId<V, S> id, std::make_signed_t<S> factor)
```

Comment:
I don't think the semantics here make sense: the best analog to `OpaqueId` is probably a pointer, so it makes sense to subtract and add and increment and `operator[]`, but not to multiply.

Reply:
OK, I see. And I understand the problem now, and it looks like [the thrust specification of sequence](https://thrust.github.io/doc/group__transformations_ga9d17b70466df6b9a57f94170935a03f8.html) is incomplete.

https://github.com/NVIDIA/thrust/blob/949ac649afc37e9f92d946f267c2390c6ab3daa7/thrust/system/detail/generic/sequence.inl#L57-L69

implicitly requires addition of another of the instance *and* multiplication of the index by a "step".

I'll think about this.

Reply:
One solution could be to have `CoreStateData::track_slots` defined as `Collection<celeritas::size_type, W, M, ThreadId>`, do the sorting/filling on numbers and we wrap it in a `TrackSlotId` when returning from `CoreTrackView::track_slot_id`

Reply:
Implementation of this idea in 0fc32cc. We could also remove the pre/post-increment operators but I'd say it still make sense to keep them if we don't care about OpaqueId being immutable.

Reply:
Actually, I like your idea and implementation. It might even be necessary when we try to use more complex algorithms (sorting). Let's stick with that.

I was thinking as an alternative to just use a kernel to set `TrackSlotId` independently for each thread, but yours is better.

Reply:
+1
I guess another way to see is that the sorting is now done via an `iterator` type which is distinct from the value type `track_slots`.